### ISSUE TITLE
Missing command to read current paragraph

### ISSUE THREAD

**Issue by adamsamec** [nvaccess/nvda#19405]
I am missing a command to read the entire paragraph for the current location of the text cursor, something like reading current line with NVDA + L, but for paragrap. For this reading of the current paragraph, I suggest a new keybaord shortcut NVDA + I.

Note that JAWS has this command, JAWSKey + Control + I.

**Comment by seanbudd** [comment]
Maybe NVDA+shift+p might be better here since `p` is used for paragraph navigation.

**Comment by CyrilleB79** [comment]
`p` / `shift+p` are used for "text paragraphs" reading, not for standard paragraph reading. I am not sure that these 2 concepts are widely distinguished by a lot of people.

But using `NVDA+shift+p` to read the current paragraph, I guess standard one, would still add to the confusion IMO. so something completely different from a gesture using `p` should be used IMO.

**Comment by adamsamec** [comment]
NVDA + P currently cycles through speech symbol levels. I suggest changing this shortcut to NVDA + Shift + P and let NVDA + P read the current paragraph instead. P and Shift + P without a modifier is used to navigate to the next or previous paragraph in the browse mode, respectively. I don't see anything confusing here.

**Comment by CyrilleB79** [comment]
@adamsamec have you clearly in mind the difference of behaviour between pressing `p` and `control+downArrow` in browse mode?

**Comment by adamsamec** [comment]
P and Shift + P in browse mode basically finds the next or previous paragraph element, possibly skipping non-paragraph elements like buttons or links, whereas Control + Down arrow or Control + Up arrow navigates even to non-paragraph elements.

Sorry, I still don't get how this relates to introducing NVDA + P as a command to read the paragraph under the cursor similar to NVDA + L for reading the current line.

**Comment by marymentzer07-prog** [comment]
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Adam ***@***.***>
Sent: Friday, December 26, 2025 12:49:38 PM
To: nvaccess/nvda ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [nvaccess/nvda] Missing command to read current paragraph (Issue #19405)

[https://avatars.githubusercontent.com/u/651076?s=20&v=4]adamsamec created an issue (nvaccess/nvda#19405)<https://github.com/nvaccess/nvda/issues/19405>

I am missing a command to read the entire paragraph for the current location of the text cursor, something like reading current line with NVDA + L, but for paragrap. For this reading of the current paragraph, I suggest a new keybaord shortcut NVDA + I.

Note that JAWS has this command, JAWSKey + Control + I.

**Comment by CyrilleB79** [comment]
Well, FWIK, there is no notion of paragraph vs non-paragraph elements at NVDA browse mode level.

Maybe I have not correctly understood your request though. Do you wish the new command to read the current paragraph, in the sense of document units jumped to with `control+up/downArrow`? Or should the new command report the current "text paragraph", in the sense of document units jumped to with `p` or `shift+p`? Or a third paragraph definition?

**Comment by adamsamec** [comment]
I think the new command  should  read the current paragraph in the sense of document units jumped to with Control + Up/Down arrow.

**Comment by CyrilleB79** [comment]
> I think the new command should read the current paragraph in the sense of document units jumped to with Control + Up/Down arrow.

So if the new command is associated to `NVDA+shift+p`, it will remind `p` / `shift+p` and not `ctrl+up/downArrow`; that's why I thought it may seem confusing.

**Comment by adamsamec** [comment]
I think it is okay. But if you have a problem with it, NVDA + I might be a good shortcut as well.

**Comment by CyrilleB79** [comment]
Personally, I have no problem with it.

My feedback was just a general one, trying to make NVDA as easy to use as possible. If my remark does not make sense for anybody else than myself, just ignore it, no problem!

**Comment by adamsamec** [comment]
If it was on me, I would assign NVDA + P to the read the current paragraph command and change the binding for the cycling between  speech symbol levels to NVDA + Shift + P or a different one. But I don't insist on this.

What is needed for this to be accepted as a feature request?

**Comment by gerald-hartig** [comment]
We don't support changing NVDA + P.

**Comment by adamsamec** [comment]
Since NVDA + L already exists for reading the current line, I thought it would be consistent to have NVDA + P to read the current paragraph without a modifier. But again, I don't insist on this.

**Comment by adamsamec** [comment]
It's been already more than half a year since i created this feature request without any progress. I just modified the issue description to mention that this proposed command would be really helpful in situations when you want to review the paragraph you are currently composing without moving the text cursor. This would make composing documents or emails easier.
